bitwarden_license/bit-web/src/app/secrets-manager/projects/projects/projects.component.ts

```diff
@@ -70,11 +82,47 @@ export class ProjectsComponent implements OnInit {
     });
   }
 
-  openDeleteProjectDialog(event: ProjectListView[]) {
-    this.dialogService.open<unknown, ProjectDeleteOperation>(ProjectDeleteDialogComponent, {
-      data: {
-        projects: event,
-      },
+  openDeleteProjectDialog(projects: ProjectListView[]) {
+    if (projects.some((project) => project.write == false)) {
+      const readOnlyProjects = projects.filter((project) => project.write == false);
+      const writeProjects = projects.filter((project) => project.write);
+      const dialogRef = this.dialogService.open<unknown, BulkConfirmationDetails>(
+        BulkConfirmationDialogComponent,
+        {
+          data: {
+            title: "deleteProjects",
+            columnTitle: "projectName",
+            message: "smProjectsDeleteBulkConfirmation",
+            details: this.getBulkConfirmationDetails(readOnlyProjects),
+          },
+        }
+      );
+
+      dialogRef.closed.pipe(takeUntil(this.destroy$)).subscribe((result) => {
+        if (result == BulkConfirmationResult.Continue) {
+          this.dialogService.open<unknown, ProjectDeleteOperation>(ProjectDeleteDialogComponent, {
+            data: {
+              projects: writeProjects,
+            },
+          });
+        }
+      });
+    } else {
+      this.dialogService.open<unknown, ProjectDeleteOperation>(ProjectDeleteDialogComponent, {
+        data: {
+          projects: projects,
```

Comment:
Fun syntax fact, you can just write it as:

```suggestion
          projects,
```